pkg/util/interrupt/child_linux.go

```diff
@@ -24,4 +24,4 @@ import (
 )
 
 // childSignals are the allowed signals that can be sent to children in Unix variant OS's
-var childSignals = []os.Signal{syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT}
```

Comment:
_linux.go is a build time convenience.  It is required.

SIGHUP would only be valid on the bsd's and linux, but not windows.

On Fri, Apr 1, 2016 at 3:33 PM, Ben Parees <notifications@github.com> wrote:

> In pkg/util/interrupt/child_linux.go
> <https://github.com/kubernetes/kubernetes/pull/23752#discussion_r58256265>
> :
>
> > @@ -24,4 +24,4 @@ import (
> >  )
> >
> >  // childSignals are the allowed signals that can be sent to children in Unix variant OS's
> > -var childSignals = []os.Signal{syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT}
>
>
> https://github.com/kubernetes/kubernetes/blob/master/pkg/util/interrupt/child.go#L25
>
> separate file (though to be honest i have no idea how we have two files in
> the same package declaring the same variable. perhaps there is some build
> magic going on to pick which file to use when building for a particular
> platform).
>
> Note that for s2i i didn't follow this pattern, not sure if it's necessary
> or not. perhaps @smarterclayton <https://github.com/smarterclayton> can
> tell us more.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23752/files/3600b16e55d76c51848ede1c4519fa13ac49859c#r58256265>
>